autoconfig.json

```diff
@@ -1,34 +1,119 @@
+        ]
+      },
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/providers",
+          "/routing/v1/peers",
+          "/routing/v1/ipns"
+        ],
+        "Write": []
+      }
+    },
+    "IPNI": {
+      "URL": "https://cid.contact",
+      "Description": "Network Indexer - content routing database for large storage providers",
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/providers"
+        ],
+        "Write": []
+      }
+    },
+    "IPNS": {
+      "URL": "https://specs.ipfs.tech/ipns/ipns-record/",
+      "Description": "InterPlanetary Name System - signed mutable records that point to CIDs with optional metadata",
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/ipns"
+        ],
+        "Write": [
+          "/routing/v1/ipns"
+        ]
+      }
+    },
+    "DNS": {
+      "URL": "https://datatracker.ietf.org/doc/html/rfc8484",
+      "Description": "DNS over HTTPS (DoH) - protocol for performing DNS resolution via the HTTPS protocol",
+      "DelegatedConfig": {
+        "Read": [
+          "/dns-query"
+        ],
+        "Write": []
+      }
+    },
+    "Example": {
+      "URL": "https://example.com",
+      "Description": "Example routing system demonstrating extensibility. Clients should ignore unknown APIs they don't recognize",
+      "DelegatedConfig": {
+        "Read": [
+          "/example/v0/read"
+        ],
+        "Write": [
+          "/example/v0/write"
+        ]
+      }
+    }
+  },
   "DNSResolvers": {
     "eth.": [
       "https://dns.eth.limo/dns-query",
       "https://dns.eth.link/dns-query"
     ]
   },
-  "DelegatedRouters": {
-    "mainnet-for-nodes-with-dht": [
-      "https://cid.contact/routing/v1/providers"
-    ],
-    "mainnet-for-nodes-without-dht": [
-      "https://delegated-ipfs.dev/routing/v1/providers",
-      "https://delegated-ipfs.dev/routing/v1/peers",
-      "https://delegated-ipfs.dev/routing/v1/ipns"
-    ]
+  "DelegatedEndpoints": {
+    "https://cid.contact": {
+      "Systems": ["IPNI"],
+      "Read": [
+        "/routing/v1/providers"
+      ],
+      "Write": []
+    },
+    "https://delegated-ipfs.dev": {
+      "Systems": ["AminoDHT", "IPNI", "IPNS", "DNS"],
+      "Read": [
+        "/routing/v1/providers",
+        "/routing/v1/peers",
+        "/routing/v1/ipns",
+        "/dns-query"
+      ],
+      "Write": [
+        "/routing/v1/ipns"
+      ]
+    },
+    "https://example.com": {
+      "Systems": ["Example"],
+      "Read": [
+        "/example/v0/read"
+      ],
+      "Write": [
+        "/example/v0/write"
+      ]
+    }
   },
-  "DelegatedPublishers": {
-    "mainnet-for-ipns-publishers-with-http": [
-      "https://delegated-ipfs.dev/routing/v1/ipns"
-    ]
+  "Profiles": {
+    "browser": {
+      "PreferDelegated": ["AminoDHT", "IPNI", "IPNS", "DNS"]
+    },
+    "daemon": {
+      "PreferNative": ["AminoDHT", "IPNS", "DNS"],
+      "PreferDelegated": ["IPNI"]
```

Comment:
I suppose it doesn't matter either way, but does it really prefer delegated, or is there just not really a native option today (yes, there's a custom API but not with a ton of added value) so we'd be forced to fallback to delegated anyway.

Reply:
Good question. 

IPNI's "native" is HTTP as well, so its mostly semantics?

The way I understand it, everyone is free to adopt IPNI's "native" APIs which (afak) are https://cid.contact/cid/ and https://cid.contact/multihash/. For example, lassie [uses `/multihash` one](https://github.com/filecoin-project/lassie/blob/8a89a36334b8f765bc4bd71a3d753b6f03cc7e29/pkg/indexerlookup/candidatesource.go#L137) :shrug: 

But for IPFS clients its mroe const-effective to use `/routing/v1` rather than use custom HTTP query code that is specific to IPNI without any benefit in doing so.